test/test.ts

```diff
@@ -0,0 +1,478 @@
+  type: 'listItem',
+  children: [paragraph],
+  position,
+  data,
+};
+
+const listItemWithChecked: ListItem = {
+  type: 'listItem',
+  children: [],
+  checked: true,
+};
+
+const listItemWithSpread: ListItem = {
+  type: 'listItem',
+  children: [],
+  spread: true,
+};
+
+const list: List = {
+  type: 'list',
+  children: [listItem],
+  position,
+  data,
+};
+
+const listWithOrdered: List = {
+  type: 'list',
+  children: [],
+  ordered: true,
+};
+
+const root: Root = {
+  type: 'root',
+  children: [],
+  position,
+  data,
+};
+
+const strong: Strong = {
+  type: 'strong',
+  children: [text],
+  position,
+  data,
+};
+
+// ## GFM
+const delete_: Delete = {
+  type: 'delete',
+  children: [text],
+  position,
+  data,
+};
+
+const footnoteDefinition: FootnoteDefinition = {
+  type: 'footnoteDefinition',
+  identifier: 'mdast',
+  children: [paragraph],
+  position,
+  data,
+};
+
+const footnoteReference: FootnoteReference = {
+  type: 'footnoteReference',
+  identifier: 'mdast',
+  position,
+  data,
+};
+
+const tableCell: TableCell = {
+  type: 'tableCell',
+  children: [text],
+  position,
+  data,
+};
+
+const tableRow: TableRow = {
+  type: 'tableRow',
+  children: [tableCell],
+  position,
+  data,
+};
+
+const table: Table = {
+  type: 'table',
+  children: [tableRow],
+  position,
+  data,
+};
+
+const tableWithAlign: Table = {
+  type: 'table',
+  align: ['left', 'center', 'right'],
+  children: [tableRow],
+  position,
+  data,
+};
+
+// ## Frontmatter
+const yaml: Yaml = {
+  type: 'yaml',
+  value: '',
+  position,
+  data,
+};
+
+// Test custom mdast node registration.
+interface Toml extends Literal {
+  type: 'toml';
+}
+
+declare module 'mdast' {
+  interface RootContentMap {
+    toml: Toml;
+  }
+
+  interface FrontmatterContentMap {
+    toml: Toml;
+  }
+}
+
+const rootOther: Root = {
+  type: 'root',
+  data,
+  position,
+  children: [{ type: 'toml', value: '' }],
+};
+
+const rootAnother: Root = {
+  type: 'root',
+  data,
+  position,
+  children: [
+    // @ts-expect-error: node not registered in `RootContentMap`.
+    { type: 'invalid' },
+  ],
+};
+
+// Register a field on `Data`, which will be available on all nodes.
+declare module 'mdast' {
+  interface Data {
+    someField?: string | undefined;
+  }
+
+  interface InlineCodeData {
+    someOtherField?: number | undefined;
+  }
+}
+
+const textWithData: Text = {
+  type: 'text',
+  value: 'value',
+  data: {
+    someField: 'a',
+    // @ts-expect-error: registered on inline codes, not on texts.
+    someOtherField: 1,
+  },
+};
+
+const textWithOtherData: Text = {
+  type: 'text',
+  value: 'value',
+  data: {
+    someField: 'a',
+    // @ts-expect-error: not registered.
+    someUnknownField: true,
+  },
+};
+
+const inlineCodeWithData: InlineCode = {
+  type: 'inlineCode',
+  value: 'value',
+  data: {
+    someField: 'a',
+    someOtherField: 1,
+    // @ts-expect-error: not registered.
+    someUnknownField: true,
+  },
+};
```

Comment:
These are taken from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mdast/index.d.ts. which should be referenced properly